emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -483,19 +489,44 @@ private void touchVmsFiles() throws MojoExecutionException {
      */
     private void saveViolationCounts() throws MojoExecutionException {
         try (Git git = Git.open(gitDir.toFile())) {
-            if (git.status().call().isClean()) {
+            if (isFunctionallyClean(git)) {
                 Files.copy(newViolationCounts, oldViolationCounts, StandardCopyOption.REPLACE_EXISTING);
 
                 try (PrintWriter out = new PrintWriter(lastShaPath.toFile())) {
                     out.println(git.getRepository().resolve(Constants.HEAD).name());
                 }
             }
-        } catch (IOException | GitAPIException ex) {
+        } catch (IOException ex) {
             ex.printStackTrace();
             throw new MojoExecutionException("Failed to save violation-counts", ex);
         }
     }
 
+    /**
+     * Whether a particular git repository is clean (there is no difference between it and its last associated commit).
+     * A repository is also considered clean if the only difference between it and the last associated commit consists
+     * only of files and directories that were created by eMOP - these are violation-counts and the .starts directory.
+     *
+     * @param git Git repository to analyze
+     * @return Boolean of whether to consider the git functionally clean for the purposes of VMS
+     * @throws MojoExecutionException if error encountered at runtime
+     */
+    private boolean isFunctionallyClean(Git git) throws MojoExecutionException {
+        try {
+            // changes in the repo will either be untracked or uncommitted - a functionally clean repo will not have
+            // any uncommitted changes but may have exactly three untracked files which were created by eMOP
+            // read more here:
+            // https://download.eclipse.org/jgit/site/6.3.0.202209071007-r/apidocs/org/eclipse/jgit/api/Status.html
+            Set<String> untracked = git.status().call().getUntracked();
+            Set<String> uncommitted = git.status().call().getUncommittedChanges();
+            return (git.status().call().isClean() || (uncommitted.size() == 0 && untracked.size() == 3
+                    && untracked.contains(".starts/last-SHA") && untracked.contains(".starts/violation-counts-old")
```

Comment:
There will be more files in `.starts` once we integrate VMS with other techniques. Is there a way to ignore the whole directory?

Reply:
Never mind, apparently there is :)